app/components/UI/Select.js

```diff
@@ -84,19 +86,53 @@ class Select extends React.PureComponent {
     items: []
   }
 
+  inputRef = React.createRef()
+
+  blurInput = () => {
+    if (this.inputRef.current) {
+      this.inputRef.current.blur()
+    }
+  }
+
   render() {
-    const { fieldApi, items, theme, ...rest } = this.props
+    let {
+      fieldApi,
+      items,
+      theme,
+      color,
+      onChange,
+      onValueSelected,
+      initialSelectedItem,
+      ...rest
+    } = this.props
+    const { setValue, setTouched } = fieldApi
+
+    let initialInputValue
+    if (initialSelectedItem) {
+      initialSelectedItem = items.find(i => i.key === initialSelectedItem)
+      initialInputValue = itemToString(initialSelectedItem)
+    }
```

Comment:
Could be simplified to
```
const findByKey = ({key}) => key === initialSelectedItem
const initialInputValue = initialSelectedItem && itemToString(items.find(findByKey))
```
to avoid `let`
But if we need to ensure `initialInputValue` equals `undefined` in case  `initialSelectedItem` is `false` `null` or empy string ternary operator should be used or just leave it as it is.

Reply:
In this case, both of these values need to be used independently and conditionally based on each other. I tried your suggestion and it doesn't produce the same results, and I tried several other things but this still seems the most straightforward and readable option to me.